cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -90,181 +133,221 @@ func newCmdApply(apf *applyPlanFlags) *cobra.Command {
 	cmd.Flags().BoolVar(&flags.renewCerts, options.CertificateRenewal, flags.renewCerts, "Perform the renewal of certificates used by component changed during upgrades.")
 	options.AddPatchesFlag(cmd.Flags(), &flags.patchesDir)
 
+	// initialize the workflow runner with the list of phases
+	applyRunner.AppendPhase(phases.NewPreflightPhase())
+	applyRunner.AppendPhase(phases.NewControlPlanePhase())
```

Comment:
there are a number of different tasks in the control plane phase PerformStaticPodUpgrade function. do we want to make them exposed as subphases in some way or do you think that's not ideal?

Reply:
A split makes sense, but I think we need to do it in steps.
Currently, `kubeadm upgrade node` have the same problem.
We can consider an independent PR for this optimization.

Reply:
can you please add a note about this in the tracking issue?

Reply:
OK